### PR TITLE
kic entrypoint: retryable update-alternatives, move earlier into execution

### PR DESCRIPTION
**Pull request by tstromberg**
Makes #8814 easier to debug - and may actually just resolve it.

Let me know if you want a separate PR to add a pristine copy of the entrypoint.

### CODE REVIEW

File: hack/images/kicbase.Dockerfile [hunk 1]
```diff
@@ -63,12 +65,12 @@ USER root
...
-  RUN echo "kic! Build: ${COMMIT_SHA} Time :$(date)" > "/kic.txt"
+    /var/cache/debconf/* \
+    /var/lib/apt/lists/* \
+    /var/log/* \
+    /tmp/* \
+    /var/tmp/* \
+    /usr/share/doc/* \
+    /usr/share/man/* \
+    /usr/share/local/* \
+    RUN echo "kic! Build: ${COMMIT_SHA} Time :$(date)" > "/kic.txt"
```
Comment (medyagh) [inline hack/images/kicbase.Dockerfile:76]: while we are at this please un-indent this RUN, this was a mistake indendation.

Comment (tstromberg) [reply hack/images/kicbase.Dockerfile:76]: Done.

**Comment by medyagh** [comment]
@tstromberg 28 failures with this new image
```
  "TestJSONOutput/serial/IncreasingCurrentSteps",
  "TestMissingContainerUpgrade",
  "TestNetworkPlugins/group/calico/Start",
  "TestNetworkPlugins/group/custom-weave/Start",
  "TestNetworkPlugins/group/kindnet/Start",
  "TestNetworkPlugins/group/auto/DNS",
  "TestNetworkPlugins/group/enable-default-cni/DNS",
  "TestNetworkPlugins/group/kubenet/DNS",
  "TestNetworkPlugins/group/bridge/DNS",
  "TestStartStop/group/crio/serial/FirstStart",
  "TestStartStop/group/containerd/serial/FirstStart",
  "TestStartStop/group/crio/serial/DeployApp",
  "TestStartStop/group/containerd/serial/DeployApp",
  "TestStartStop/group/crio/serial/SecondStart",
  "TestStartStop/group/containerd/serial/SecondStart",
  "TestStartStop/group/crio/serial/UserAppExistsAfterStop",
  "TestStartStop/group/containerd/serial/UserAppExistsAfterStop",
  "TestStartStop/group/crio/serial/AddonExistsAfterStop",
  "TestStartStop/group/containerd/serial/AddonExistsAfterStop",
  "TestStartStop/group/crio/serial/VerifyKubernetesImages",
  "TestStartStop/group/crio/serial/Pause",
  "TestStartStop/group/containerd/serial/VerifyKubernetesImages",
  "TestStartStop/group/containerd/serial/Pause"
]
-------------------------------------------------------
*** 28 Failed ***
```

**Comment by medyagh** [comment]
for cio first start:

```
failed to set bridge addr: could not add IP address to "cni0": permission denied
```

**Comment by medyagh** [comment]
for containerd first start I see
```
I0724 22:19:37.408877  335425 exit.go:58] WithError(failed to start node)=startup failed: wait 6m0s for node: waiting for apps_running: expected k8s-apps: missing components: kube-dns called from:
	goroutine 1 [running]:
	runtime/debug.Stack(0x0, 0x0, 0x100000000000000)
		/home/jenkins/actions-runner/_work/_tool/go/1.14.6/x64/src/runtime/debug/stack.go:24 +0x9d
	k8s.io/minikube/pkg/minikube/exit.WithError(0x1ba7c56, 0x14, 0x1ebf200, 0xc00090f1e0)
		/home/jenkins/actions-runner/_work/minikube/minikube/pkg/minikube/exit/exit.go:58 +0x34
	k8s.io/minikube/cmd/minikube/cmd.runStart(0x2cd0820, 0xc0003f6620, 0x2, 0xe)
		/home/jenkins/actions-runner/_work/minikube/minikube/cmd/minikube/cmd/start.go:206 +0x505
	github.com/spf13/cobra.(*Command).execute(0x2cd0820, 0xc0003f6540, 0xe, 0xe, 0x2cd0820, 0xc0003f6540)
		/home/jenkins/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:846 +0x29d
	github.com/spf13/cobra.(*Command).ExecuteC(0x2ccf860, 0x0, 0x1, 0xc000a1a5f0)
		/home/jenkins/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x349
	github.com/spf13/cobra.(*Command).Execute(...)
		/home/jenkins/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
	k8s.io/minikube/cmd/minikube/cmd.Execute()
		/home/jenkins/actions-runner/_work/minikube/minikube/cmd/minikube/cmd/root.go:106 +0x72c
	main.main()
		/home/jenkins/actions-runner/_work/minikube/minikube/cmd/minikube/main.go:71 +0x11f
	W0724 22:19:37.409094  335425 out.go:249] failed to start node: startup failed: wait 6m0s for node: waiting for apps_running: expected k8s-apps: missing components: kube-dns
	* 
	X failed to start node: startup failed: wait 6m0s for node: waiting for apps_running: expected k8s-apps: missing components: kube-dns
	* 
	* minikube is exiting due to an error. If the above message is not useful, open an issue:
	  - https://github.com/kubernetes/minikube/issues/new/choose
** /stderr **
```

**Pull request closed without merging**